libs/related-products/state/src/reducers/related-products/reducer.ts

```diff
@@ -0,0 +1,31 @@
+import { DaffProduct } from '@daffodil/product';
+import {
+  DaffProductPageActions,
+  DaffProductPageActionTypes,
+} from '@daffodil/product/state';
+
+import { DaffRelatedProductsReducerState } from './reducer-state.interface';
+
+/**
+ * Initial values of the related product state.
+ */
+export const initialState: DaffRelatedProductsReducerState = {
+  relatedProductIds: [],
+};
+
+/**
+ * Reducer function that catches actions and changes/overwrites related product state.
+ */
+export function daffRelatedProductsReducer<T extends DaffProduct>(state = initialState, action: DaffProductPageActions<T>): DaffRelatedProductsReducerState {
+  switch (action.type) {
+    case DaffProductPageActionTypes.ProductPageLoadSuccessAction:
+      const product = action.payload.products.filter(({ id }) => id === action.payload.id)[0];
+
+      return {
+        ...state,
+        relatedProductIds: product.related?.map(({ id }) => id) || [],
```

Comment:
won't the related products already be removed from the base product at this point, since the meta reducers run before this reducer?

Reply:
the meta reducer is immutable (see the test case) and only the main product reducers should be affected by the provided meta reducers

Reply:
> the meta reducer is immutable
But the related product reducer uses the action to set its state, not existing state in `@daffodil/product`.

> only the main product reducers should be affected by the provided meta reducers
Ok, that makes sense